homeassistant/components/transmission/__init__.py

```diff
@@ -397,42 +397,55 @@ def init_torrent_list(self):
 
     def check_completed_torrent(self):
         """Get completed torrent functionality."""
-        current_completed_torrents = [
-            torrent for torrent in self._torrents if torrent.status == "seeding"
-        ]
-        freshly_completed_torrents = set(current_completed_torrents).difference(
-            self._completed_torrents
-        )
-        self._completed_torrents = current_completed_torrents
+        old_completed_torrent_names = {
+            torrent.name for torrent in self._completed_torrents
+        }
 
-        for torrent in freshly_completed_torrents:
-            self.hass.bus.fire(
-                EVENT_DOWNLOADED_TORRENT, {"name": torrent.name, "id": torrent.id}
-            )
+        current_completed_torrents = []
+        current_completed_torrent_names = set()
```

Comment:
Where is the set used besides below where we add items to it?

Reply:
Oops don't need that anymore!  I will change it to comprehensions then.